app/controllers/accounts_controller.rb

```diff
@@ -31,9 +31,22 @@ def create
     redirect_to my_account_path
   end
 
+  def archive
+    return redirect_to account_path unless accessed?
+
+    account.update!(archived_at: Time.current)
+    AccountShare.where(account_id: account).delete_all
```

Comment:
Why this is needed? We should change only the account here. All other related data, if it affects the system, should definitely be disabled. But it should be disabled only by filtering, e.g. checking if the related account is not archived.